File: samples/server/petstore/springboot/src/main/java/org/openapitools/model/OuterEnumIntegerDefaultValue.java
package org.openapitools.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonValue;
import org.openapitools.jackson.nullable.JsonNullable;
import javax.validation.Valid;
import javax.validation.constraints.*;

import com.fasterxml.jackson.annotation.JsonCreator;

/**
 * Gets or Sets OuterEnumIntegerDefaultValue
 */
public enum OuterEnumIntegerDefaultValue {
  
  NUMBER_0(0),
  
  NUMBER_1(1),
  
  NUMBER_2(2);

  private Integer value;

  OuterEnumIntegerDefaultValue(Integer value) {
    this.value = value;
  }

  @Override
  @JsonValue
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static OuterEnumIntegerDefaultValue fromValue(Integer value) {
    for (OuterEnumIntegerDefaultValue b : OuterEnumIntegerDefaultValue.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }
}

File: samples/server/petstore/kotlin-springboot-request-cookie/src/main/kotlin/org/openapitools/model/OuterEnumIntegerDefaultValue.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonValue
import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.DecimalMax
import jakarta.validation.constraints.DecimalMin
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size
import jakarta.validation.Valid
import io.swagger.v3.oas.annotations.media.Schema

/**
* 
* Values: _0,_1,_2
*/
enum class OuterEnumIntegerDefaultValue(val value: kotlin.Int) {

    @JsonProperty(0) _0(0),
    @JsonProperty(1) _1(1),
    @JsonProperty(2) _2(2)
}

